pkg/services/ngalert/notifier/channels/discord.go

```diff
@@ -96,69 +134,78 @@ func newDiscordNotifier(fc channels.FactoryConfig) (*DiscordNotifier, error) {
 func (d DiscordNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 	alerts := types.Alerts(as...)
 
-	bodyJSON := simplejson.New()
+	msg := discordMessage{}
 
 	if !d.settings.UseDiscordUsername {
-		bodyJSON.Set("username", "Grafana")
+		msg.Username = "Grafana"
 	}
 
 	var tmplErr error
 	tmpl, _ := channels.TmplText(ctx, d.tmpl, as, d.log, &tmplErr)
 
-	bodyJSON.Set("content", tmpl(d.settings.Message))
+	msg.Content = tmpl(d.settings.Message)
 	if tmplErr != nil {
 		d.log.Warn("failed to template Discord notification content", "error", tmplErr.Error())
 		// Reset tmplErr for templating other fields.
 		tmplErr = nil
 	}
+	truncatedMsg, truncated := channels.TruncateInRunes(msg.Content, discordMaxMessageLen)
+	if truncated {
+		key, err := notify.ExtractGroupKey(ctx)
+		if err != nil {
+			return false, err
+		}
+		d.log.Warn("Truncated content", "key", key, "max_runes", discordMaxMessageLen)
+		msg.Content = truncatedMsg
+	}
 
 	if d.settings.AvatarURL != "" {
-		bodyJSON.Set("avatar_url", tmpl(d.settings.AvatarURL))
+		msg.AvatarURL = tmpl(d.settings.AvatarURL)
 		if tmplErr != nil {
 			d.log.Warn("failed to template Discord Avatar URL", "error", tmplErr.Error(), "fallback", d.settings.AvatarURL)
-			bodyJSON.Set("avatar_url", d.settings.AvatarURL)
+			msg.AvatarURL = d.settings.AvatarURL
 			tmplErr = nil
 		}
 	}
 
-	footer := map[string]interface{}{
-		"text":     "Grafana v" + d.appVersion,
-		"icon_url": "https://grafana.com/static/assets/img/fav32.png",
+	footer := &discordFooter{
+		Text:    "Grafana v" + d.appVersion,
+		IconURL: "https://grafana.com/static/assets/img/fav32.png",
 	}
 
-	linkEmbed := simplejson.New()
+	linkEmbed := discordLinkEmbed{}
```

Comment:
```suggestion
	var linkEmbed discordLinkEmbed
```

Reply:
Or directly adding `Footer` and `Type` values in the same declaration could be an option too.